docs/long_desc.rst

```diff
@@ -3,11 +3,33 @@ BanditPAM: A state-of-the-art, high-performance k-medoids algorithm
 
 Introduction
 ------------
-Clustering algorithms such as k-means are ubiquitous in modern data science applications. Despite its popularity, however, k-means has several drawbacks. K-means only supports certain distance metrics, and its cluster centers can lack interpretability. Alternatively, k-medoids uses the points in the dataset itself -- the medoids -- as cluster centers. This enables interpretability of the cluster centers. Furthermore, k-medoids supports arbitrary dissimilarity functions in place of a distance metric.
+Clustering algorithms such as k-means are ubiquitous in modern data science applications. 
+Despite its popularity, however, k-means has several drawbacks. Firstly, k-means only supports certain distance metrics, 
+while different distance metrics may be desirable in other applications. For example, L1 norm and cosine distance are often used in
+recommendation systems and single-cell RNA-seq analysis. Secondly, the cluster center of k-means is in general not a point 
+in the dataset and hence lack of interpretability. This is problemetic when the problem requires interpretability on the center. 
+For example, when the data is structured, such as images in computer vision where the mean image is virtually random noise.
+Alternatively, k-medoids uses the points in the dataset itself -- the medoids -- as cluster centers. 
+This enables interpretability of the cluster centers. Furthermore, k-medoids supports arbitrary dissimilarity functions 
+in place of a distance metric.
 
 Details
 -------
-In this package, we provide a high-performance implementation of BanditPAM , a state-of-the-art k-medoids algorithm. BanditPAM matches state-of-the-art in clustering quality but improves the runtime of previous approaches from O(n^2) to O(nlogn) in each iteration. Furthermore, BanditPAM supports arbitrarily dissimilarity functions between points -- these functions need not even be metrics. This implementation is written in C++ for performance, but is callable from Python via Python bindings. For further details, please see the paper at https://arxiv.org/abs/2006.06856 for the full paper and the code at https://github.com/ThrunGroup/BanditPAM/blob/main/README.md
+Despite its advantages, k-medoids clustering is less popular than k-means due to its computational cost. 
+The current state-of-the-art k-medoids algorithms PAM and FastPAM scale quadratically in the dataset size in each iterations
+but they are still significantly slower than k-means, which scale linearly in dataset size in each iteration.
+In this package, we provide a high-performance implementation of BanditPAM, 
+a state-of-the-art k-medoids algorithm. The main contributions of BanditPAM are as follows:
+
+* BanditPAM matches state-of-the-art in clustering quality but improves the runtime of previous approaches from O(n^2) to O(nlogn) in each iteration. 
```

Comment:
`BanditPAM matches the best prior work in clustering quality but scales as O(nlogn) instead of O(n^2) in each iteration.`